File: chwazifake_android/app/src/main/java/com/hkwg/chwazifake/domain/repository/CircleRepository.kt
package com.hkwg.chwazifake.domain.repository

import com.hkwg.chwazifake.domain.model.CircleState

interface CircleRepository {
    fun addCircle(circle: CircleState)
    fun removeCircle(circle: CircleState)
    fun updateCircle(circle: CircleState)
    fun getCircles(): List<CircleState>
} 

File: chwazifake_flutter/lib/domain/repository/circle_repository.dart
import '../model/circle_state.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

final circleRepositoryProvider = Provider<CircleRepository>((ref) {
  return CircleRepositoryImpl();
});

abstract class CircleRepository {
  List<CircleState> getCircles();
  void addCircle(CircleState circle);
  void removeCircle(CircleState circle);
  void updateCircle(CircleState circle);
}

class CircleRepositoryImpl implements CircleRepository {
  final List<CircleState> _circles = [];

  @override
  List<CircleState> getCircles() => _circles;

  @override
  void addCircle(CircleState circle) {
    _circles.add(circle);
  }

  @override
  void removeCircle(CircleState circle) {
    _circles.remove(circle);
  }

  @override
  void updateCircle(CircleState circle) {
    // Implementation needed
  }
} 